keras_cv/ops/box_matcher.py

```diff
@@ -0,0 +1,235 @@
+        force_match_for_each_col: bool = False,
+    ):
+        """Constructs ArgmaxBoxMatcher.
+
+        Args:
+          thresholds: A sorted list of floats to classify the matches into
+            different results (e.g. positive or negative or ignored match). The
+            list will be prepended with -Inf and and appended with +Inf.
+          match_values: A list of integers representing matched results (e.g.
+            positive or negative or ignored match). len(`match_values`) must
+            equal to len(`thresholds`) + 1.
+          force_match_for_each_col: each row will be argmax matched to at
+            least one column. This means some columns will be matched to
+            multiple rows while some columns will not be matched to any rows.
+            Filtering by `thresholds` will make less columns match to positive
+            result. Setting this to True guarantees that each column will be
+            matched to positive result to at least one row.
+
+        Raises:
+          ValueError: if `thresholds` not sorted or
+            len(`match_values`) != len(`thresholds`) + 1
+        """
+        if not all([lo <= hi for (lo, hi) in zip(thresholds[:-1], thresholds[1:])]):
+            raise ValueError("`threshold` must be sorted, got {}".format(thresholds))
+        self.match_values = match_values
+        if len(match_values) != len(thresholds) + 1:
+            raise ValueError(
+                "len(`match_values`) must be len(`thresholds`) + 1, got "
+                "match_values {}, thresholds {}".format(match_values, thresholds)
+            )
+        thresholds = thresholds[:]
+        thresholds.insert(0, -float("inf"))
+        thresholds.append(float("inf"))
+        self.thresholds = thresholds
+        self._force_match_for_each_col = force_match_for_each_col
+
+    def __call__(self, similarity_matrix: tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]:
+        """Matches each row to a column based on argmax
+
+        Args:
+          similarity_matrix: A float Tensor of shape [num_rows, num_cols] or
+            [batch_size, num_rows, num_cols] representing any similarity metric.
+
+        Returns:
+          matched_columns: An integer tensor of shape [num_rows] or [batch_size,
+            num_rows] storing the index of the matched colum for each row.
+          matched_values: An integer tensor of shape [num_rows] or [batch_size,
+            num_rows] storing the match result (positive match, negative match,
+            ignored match).
+        """
+        squeeze_result = False
+        if len(similarity_matrix.shape) == 2:
+            squeeze_result = True
+            similarity_matrix = tf.expand_dims(similarity_matrix, axis=0)
+        static_shape = similarity_matrix.shape.as_list()
+        num_rows = static_shape[1] or tf.shape(similarity_matrix)[1]
+        batch_size = static_shape[0] or tf.shape(similarity_matrix)[0]
+
+        def _match_when_cols_are_empty():
```

Comment:
should this be _math_when_rows_are_empty()?

Reply:
it should be cols -- when gt_boxes are empty.